ContactDetailsApi/V2/Infrastructure/IPWhitelistMiddleware.cs

```diff
@@ -0,0 +1,62 @@
+using ContactDetailsApi.V2.Domain;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace ContactDetailsApi.V2.Infrastructure
+{
+    public class IPWhitelistMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger<IPWhitelistMiddleware> _logger;
+        private readonly HashSet<string> _safelist;
+
+        public IPWhitelistMiddleware(
+            RequestDelegate next,
+            ILogger<IPWhitelistMiddleware> logger)
+        {
+            _next = next;
+            _logger = logger;
+            try
+            {
+                var safelist = Environment.GetEnvironmentVariable("WHITELIST_IP_ADDRESS");
+                _logger.LogInformation("whitelist ip address is {safelist}", safelist);
+                var ips = safelist.Split(';');
+                _safelist = new HashSet<string>(ips);
+
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation($" cannot get env var {ex.Message}");
+            }
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (context.Request.Path == "/api/v2/servicesoft/contactDetails")
```

Comment:
I see... so this middleware is on the API level.

If I recall well, you should be able to create a middleware that's applicable as an attribute `[IPWhitelistAttribure]` against the Controller method, so:

```cs
 [HttpGet]
 [Route("contactDetails")]
 [IPWhitelist("SERVICE_SOFT_IP_WHITELIST")]
 public async Task<IActionResult> FetchAllContactDetailsByUprn()
 {
     // endpoint doing its work...
 }
```

This way all this logic will not need to be called for every Hackney's internal request.
Plus. you'll be able to omit this Path check.

Reply:
This!